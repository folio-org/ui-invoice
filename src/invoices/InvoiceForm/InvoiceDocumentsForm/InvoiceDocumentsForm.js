import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import {
  FieldArray,
  arrayPush,
} from 'redux-form';
import { FormattedMessage } from 'react-intl';

import {
  Col,
  Row,
  Label,
} from '@folio/stripes/components';
import { FileUploader } from '@folio/stripes-acq-components';

import InvoiceDocumentFields from './InvoiceDocumentFields';

const InvoiceDocumentsForm = ({ dispatch }) => {
  const selectFile = useCallback(
    (file) => {
      const reader = new FileReader();

      reader.onload = (event) => {
        dispatch(arrayPush('invoiceForm', 'documents', {
          name: file.name,
          data: event.target.result,
        }));
      };

      reader.readAsDataURL(file);
    },
    [dispatch],
  );

  return (
    <Row>
      <Col xs={4}>
        <Label>
          <FormattedMessage id="ui-invoice.invoice.documents.title" />
        </Label>

        <FieldArray
          component={InvoiceDocumentFields}
          name="documents"
        />
      </Col>

      <Col xs={8}>
        <FileUploader
          onSelectFile={selectFile}
        />
      </Col>
    </Row>
  );
};

InvoiceDocumentsForm.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default InvoiceDocumentsForm;
