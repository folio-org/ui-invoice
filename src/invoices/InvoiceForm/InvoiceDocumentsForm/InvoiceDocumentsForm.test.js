import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import stripesFinalForm from '@folio/stripes/final-form';

import InvoiceDocumentsForm from './InvoiceDocumentsForm';

const TestForm = stripesFinalForm({})(
  (props) => {
    return (
      <form>
        <InvoiceDocumentsForm {...props} />
      </form>
    );
  },
);

const renderForm = ({
  isNonInteractive = false,
  initialValues = {},
  push = jest.fn(),
} = {}) => render(
  <TestForm
    onSubmit={jest.fn()}
    initialValues={initialValues}
    isNonInteractive={isNonInteractive}
    push={push}
  />,
  { wrapper: MemoryRouter },
);

describe('InvoiceDocumentsForm', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure with defined adjustments', async () => {
    const { asFragment } = renderForm({
      initialValues: {
        documents: [{ id: 'documentId', name: 'Invoice document' }, { name: 'Document link' }],
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
