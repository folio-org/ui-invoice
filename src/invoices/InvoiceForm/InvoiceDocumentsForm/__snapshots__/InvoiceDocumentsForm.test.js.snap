// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceDocumentsForm should render correct structure with defined adjustments 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-4"
      >
        <label
          class="label"
        >
          ui-invoice.invoice.documents.title
        </label>
        <div
          class="row"
          data-test-invoice-form-document="true"
        >
          <div
            class="col-xs-12"
          >
            <div
              class="invoiceDocumentField"
            >
              <div
                class="invoiceDocumentFieldContent invoiceDocumentFieldContentEven"
              >
                <button
                  class="invoiceDocumentButton"
                  type="button"
                >
                  Invoice document
                </button>
              </div>
              <button
                aria-label="trash"
                class="iconButton medium"
                data-test-remove-document-button="true"
                icon="trash"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset"
                  data-test-remove-document-button="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="row"
          data-test-invoice-form-document="true"
        >
          <div
            class="col-xs-12"
          >
            <div
              class="invoiceDocumentField"
            >
              <div
                class="invoiceDocumentFieldContent invoiceDocumentFieldContentOdd"
              >
                <span>
                  Document link
                </span>
              </div>
              <button
                aria-label="trash"
                class="iconButton medium"
                data-test-remove-document-button="true"
                icon="trash"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset"
                  data-test-remove-document-button="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xs-8"
      >
        <div
          tabindex="0"
        >
          <div
            class="fileUploader"
          >
            <input
              autocomplete="off"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="fileUploaderTitle"
            >
              stripes-acq-components.fileUploader.dropFile
            </div>
            <button
              class="button primary"
              type="button"
            >
              <span
                class="inner"
              >
                stripes-acq-components.fileUploader.selectFile
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;
